src/Microsoft.TemplateEngine.Cli/Commands/create/TemplateOption.cs

```diff
@@ -5,24 +5,94 @@
 
 namespace Microsoft.TemplateEngine.Cli.Commands
 {
-    internal class TemplateOption
+    internal class TemplateOption : IEquatable<TemplateOption>
     {
-        private readonly Option _option;
+        private Option _option;
 
         internal TemplateOption(
             CliTemplateParameter parameter,
-            IReadOnlyList<string> aliases)
+            IReadOnlySet<string> aliases)
         {
             TemplateParameter = parameter;
             Aliases = aliases;
-            _option = parameter.GetOption(aliases);
+            _option = TemplateParameter.GetOption(Aliases);
         }
 
         internal CliTemplateParameter TemplateParameter { get; private set; }
 
-        internal IReadOnlyList<string> Aliases { get; private set; }
+        internal IReadOnlySet<string> Aliases { get; private set; }
 
         internal Option Option => _option;
 
+        public bool Equals(TemplateOption? other)
+        {
+            if (other == null)
+            {
+                return false;
+            }
+
+            if (!string.Equals(TemplateParameter.Name, other.TemplateParameter.Name, StringComparison.Ordinal))
+            {
+                return false;
+            }
+
+            if (TemplateParameter.Type != other.TemplateParameter.Type)
+            {
+                return false;
+            }
+
+            if (Aliases.Count != other.Aliases.Count)
+            {
+                return false;
+            }
+
+            foreach (string alias in other.Aliases)
+            {
+                if (!Aliases.Contains(alias))
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        public override bool Equals(object? obj)
+        {
```

Comment:
nit: this can be simplified (https://docs.microsoft.com/en-us/visualstudio/ide/reference/generate-equals-gethashcode-methods?view=vs-2022):

```
=> Equals(obj as TemplateOption);
```